### ISSUE TITLE
Framework occasionally replaces files with modified versions

### ISSUE THREAD

**Issue by itcreator** [goaop/framework#224]
before first run go-aop-php replaces vendor/autoload.php

from some like this

```
<?php

// autoload.php @generated by Composer

require_once __DIR__ . '/composer' . '/autoload_real.php';

return ComposerAutoloaderInit798bcdb19f45fb3fe1a5432366b6f53e::getLoader();

```

to

```
// autoload.php @generated by Composer

require_once \Go\Instrument\Transformer\FilterInjectorTransformer::rewrite( '/vagrant/vendor' . '/composer' . '/autoload_real.php', '/vagrant/vendor');

return ComposerAutoloaderInit798bcdb19f45fb3fe1a5432366b6f53e::getLoader();

```

And I see this error

```
PHP Fatal error:  Class 'Go\Instrument\Transformer\FilterInjectorTransformer' not found in /vagrant/vendor/autoload.php on line 5
PHP Stack trace:
PHP   1. {main}() /vagrant/bin/console.php:0
PHP   2. require_once() /vagrant/bin/console.php:4

```

and go-aop php replace directives require/include in other libs

for example Symfony\Component\Translation\Translator::initializeCacheCatalogue():373

before: 

```
$this->catalogues[$locale] = include $cache;
```

after:

```
$this->catalogues[$locale] = include \Go\Instrument\Transformer\FilterInjectorTransformer::rewrite( $cache, '/vagrant/vendor/symfony/translation/Symfony/Component/Translation');
```

It is correct?

**Comment by lisachenko** [comment]
Hello, sorry for the late response, there are a lot of work to do on my job )

Your description is looks similar to the #181 and I know about it (when `cacheDir` is not under `appDir`). This should be more stable in the master branch,aka `1.0-dev`.

Please, report me your version of framework, as well, with your configuration for the kernel.

**Comment by itcreator** [comment]
latest stable version 0.6.0

```
class MyAspectKernel extends AspectKernel
{
    /**
     * Configure an AspectContainer with advisors, aspects and pointcuts
     *
     * @param AspectContainer $container
     *
     * @return void
     */
    protected function configureAop(AspectContainer $container)
    {
        $container->registerAspect(new MyWatcher());
    }
```

autoloading:

```
use \App\MyAspectKernel;

include __DIR__ . '/src/App/MyAspectKernel.php';
// Prevent an error about nesting level
ini_set('xdebug.max_nesting_level', 500);
// Initialize an application aspect container
$applicationAspectKernel = StatusAspectKernel::getInstance();
$applicationAspectKernel->init([
    'debug' => true,
    // Application root directory
    'appDir'   => __DIR__ . '/src' ,
    // Cache directory
    'cacheDir' => '/var/cache/myApp/aosd',
//    'includePaths' => [],
    'excludePaths' => array(
        __DIR__ . '/vendor'
    )
]);
```

And this problem is reproduced on the version 1.0.0.alpha

**Comment by lisachenko** [comment]
Ah, i was right, your `cacheDir` is not under `appDir`, this is an edge case. However, I was sure, that this should be fixed in the master branch, I'll recheck that one more time to be sure.

**Comment by itcreator** [comment]
Thanks for response.

Can you describe this case on README.md?

**Comment by lisachenko** [comment]
I'm consider this issue as a bug, so this should be fixed for 1.0.0 release. But for now I recommend to put a cache into the application directory.

**Comment by tinogo** [comment]
Hello,

we have a very similar problem.
We are using the Zend Framework 1 with some Symfony Components (especially the Symfony-DI-Container).
All external libraries are included via composer but the ZF1, which is included "old-school" via the PHP include_path.
When the Symfony DI dumps it's container and therefore initializes the Go AOP Framework, it is modifing the "require" and "require_once" statements of the original ZF1 components (it is overwriting them)
Is there any solution for this?

If I include the ZF1 via composer and remove it from the include_path, all is working well, but we can't make the switch, as it's a big legacy application and can't foresee the possible side effects...

Do you have any solution to fix this?

FYI: We are using the current stable version 0.6.0.

**Comment by lisachenko** [comment]
Should be fixed in 0.6.1 and 1.0.0 as well.

**Comment by kukoman** [comment]
Hi all,

did you guys managed to run it with ZF1 mvc? It works fine with my custom code but I can't really intercept any Controller::action

``` php
class Test_IndexController extends My\AbstractController
{
    public function indexAction($test = null)
    {
```

AbstractController

``` php
    public function dispatch($action)
    {
        $actionParams = $this->getActionParams($action);
        $this->$action(...$actionParams);
        $this->_helper->notifyPostDispatch();
    }
```

my aspect:

``` php
    /**
     * Method that should be called before real method
     *
     * @param MethodInvocation $invocation Invocation
     * @After("execution(public|protected|private **->*Action(*))")
     */
    public function beforeMethodExecution(MethodInvocation $invocation)
    {
```

or am I doing something stupid here?

using the latest libs from master with latest php7

many thanks
